src/machines/child_workflow_state_machine.rs

```diff
@@ -111,3 +340,487 @@ pub(super) struct Terminated {}
+            })
+        }
+    }
+
+    #[fixture]
+    fn child_workflow_happy_hist() -> ManagedWFFunc {
+        let func = WorkflowFunction::new(parent_wf);
+        let t = canned_histories::single_child_workflow("child-id-1");
+        assert_eq!(3, t.get_full_history_info().unwrap().wf_task_count());
+        ManagedWFFunc::new(t, func, vec![[Expectation::Success as u8].into()])
+    }
+
+    #[fixture]
+    fn child_workflow_fail_hist() -> ManagedWFFunc {
+        let func = WorkflowFunction::new(parent_wf);
+        let t = canned_histories::single_child_workflow_fail("child-id-1");
+        assert_eq!(3, t.get_full_history_info().unwrap().wf_task_count());
+        ManagedWFFunc::new(t, func, vec![[Expectation::Failure as u8].into()])
+    }
+
+    #[fixture]
+    fn child_workflow_start_fail_hist() -> ManagedWFFunc {
+        let func = WorkflowFunction::new(parent_wf);
+        let t = canned_histories::single_child_workflow_start_fail("child-id-1");
+        assert_eq!(2, t.get_full_history_info().unwrap().wf_task_count());
+        ManagedWFFunc::new(t, func, vec![[Expectation::StartFailure as u8].into()])
+    }
+
+    async fn parent_wf(mut ctx: WfContext) -> WorkflowResult<()> {
+        let expectation = Expectation::try_from_u8(ctx.get_args()[0].data[0]).unwrap();
+        let req = StartChildWorkflowExecution {
+            workflow_id: "child-id-1".to_string(),
+            workflow_type: "child".to_string(),
+            input: vec![],
+            ..Default::default()
+        };
+        match (expectation.clone(), ctx.start_child_workflow(req).await) {
+            (Expectation::Success | Expectation::Failure, StartStatus::Succeeded(_)) => {}
+            (Expectation::StartFailure, StartStatus::Failed(_)) => return Ok(().into()),
+            _ => return Err(anyhow!("Unexpected start status")),
+        };
+        match (
+            expectation,
+            ctx.child_workflow_result("child-id-1".to_string())
+                .await
+                .status,
+        ) {
+            (Expectation::Success, Some(child_workflow_result::Status::Completed(_))) => {
+                Ok(().into())
+            }
+            (Expectation::Failure, _) => Ok(().into()),
+            _ => Err(anyhow!("Unexpected child WF status")),
+        }
+    }
+
+    #[rstest(
+        wfm,
+        case::success(child_workflow_happy_hist()),
+        case::failure(child_workflow_fail_hist())
+    )]
+    #[tokio::test]
+    async fn single_child_workflow_until_completion(mut wfm: ManagedWFFunc) {
+        wfm.get_next_activation().await.unwrap();
+        let commands = wfm.get_server_commands().await.commands;
+        assert_eq!(commands.len(), 1);
+        assert_eq!(
+            commands[0].command_type,
+            CommandType::StartChildWorkflowExecution as i32
+        );
+
+        wfm.get_next_activation().await.unwrap();
+        let commands = wfm.get_server_commands().await.commands;
+        assert_eq!(commands.len(), 0);
```

Comment:
What happens during this activation? I would expect this one to not exist - the first starting, and the second resolving, seems like it should be all that is needed?

Reply:
no because there's an activation just notifying that Child WF has started.
In the case of the WF above we do not generate any commands between started and complete.

Reply:
Add a little comment that explains this please, otherwise makes  sense